rhp/v3/encoding.go

```diff
@@ -323,6 +327,10 @@ func (r *RPCExecuteProgramResponse) DecodeFrom(d *types.Decoder) {
 	}
 	(*types.V1Currency)(&r.TotalCost).DecodeFrom(d)
 	(*types.V1Currency)(&r.FailureRefund).DecodeFrom(d)
+	if r.OutputLength > maxOutputLength {
+		d.SetErr(fmt.Errorf("output length %d exceeds maximum of 10 MiB", r.OutputLength))
+		return
+	}
```

Comment:
I wonder if you could do `r.OutputLength = uint64(d.ReadPrefix())`? That would automatically limit the size to the number of undecoded bytes remaining.